source/vibe/internal/async.d

```diff
@@ -138,15 +138,15 @@ void asyncAwaitAny(bool interruptible, Waitables...)(string func = __FUNCTION__)
 				};
 
 				debug(VibeAsyncLog) logDebugV("Starting operation %%s", %1$s);
-				alias WR%1$s = typeof(Waitables[%1$s].wait(callback_%1$s));
-				static if (is(WR%1$s == void)) Waitables[%1$s].wait(callback_%1$s);
-				else auto wr%1$s = Waitables[%1$s].wait(callback_%1$s);
+				alias WR%1$s = typeof(Waitables[%1$s].wait(() @trusted { return callback_%1$s; } ()));
+				static if (is(WR%1$s == void)) Waitables[%1$s].wait(() @trusted { return callback_%1$s; } ());
+				else auto wr%1$s = Waitables[%1$s].wait(() @trusted { return callback_%1$s; } ());
```

Comment:
Could this end up trusting user code ?

Reply:
I don't think so, the callbacks here are all confined to vibe-core's own source code and any user code/callbacks are layered on top.